src/overview/reducer.js

```diff
@@ -15,7 +16,19 @@ function setSearchResult(state, {searchResult}) {
     return {...state, searchResult}
 }
 
+function showLoadIndicator(state){
+	var temp = state.waitingForResults+1
```

Comment:
Still doesn't feel the right way to keep immutability. Is there any other way to do this? Something a bit more elegant?